AK/JsonParser.cpp

```diff
@@ -275,6 +277,10 @@ ErrorOr<JsonValue> JsonParser::parse_number()
         }
 #ifndef KERNEL
     }
+
+    auto as_double = value.to_number<double>();
+    if (sign == -1 && as_double == 0)
+        value = JsonValue(abs(as_double) * sign);
```

Comment:
This still seems overly complicated. Can't we just check if `number_string == "-0"sv` at the right point (probably an else if after the is_double branch) and return a negative zero for that case?

Reply:
A negative zero could be written a number of ways, `-0`, `-0.0`, `-0.00`, ect. No combination of equality and/or `starts_with` would be able to handle this, instead it would have to perform an ad-hoc byte by byte parse to check. Still a potential solution though.